readme/end-to-end-testing.md

```diff
@@ -1,25 +1,103 @@
+Playwright starts a server with the modified code, takes full-page screenshots, and compares them with reference images. Discrepancies result in test failures, with comparison results saved in the `src/__tests__/visualTesting/results` folder, helping identify and fix issues.
 
-## In case the site has undergone intentional changes
+## ⚙️Installation, Configuring and Preparing
 
-If the site has undergone intentional changes in appearance, and the developer getting an error realizes that the changes in the design are not a bug, it is enough for the developer to delete the folder with reference samples (for example, `src/__tests__/visualTesting/navbar.spec.ts-snapshots`), and run end-to-end testing using the command `npx playwright test`.
+- make sure that all dependencies are installed:
 
-When you first start testing without samples, the library will create them independently.
+```
+npm install
+```
 
-## For ease of testing
+- ⚠️ if you use Playwright for the first time, installing browser's engines is required:
 
-For ease of testing and checking its results, the library has a user interface, which can be launched with the command `npx playwright test --ui`.
+```
+npx playwright install
+```
 
-<img src="./assets/UI.jpg" alt="image">
+- ⚠️ make sure Vite develop server is able to use `localhost:5173` port. On start Playwright runs dev server automatically and then goes to the hardcoded http://localhost:5173 page.
+
+There are no any other configurations and preparations for testing.
+
+## 📝Typical Use Cases for Beginners
+
+### 🕵Detecting Visual Bugs Using User Interface
+
+- run
+
+```
+npx playwright test --ui
+```
+
+- choose browsers for testing and run any test you need separately or all the tests at once
+
+<img src="./assets/end-to-end-testing/UI.jpg" alt="image">
+
+- if there are any problems whith screenshots comparison, go to `src/__tests__/visualTesting/results` and watch the results like this:
+
+<img src="./assets/end-to-end-testing/origin-diff.png" alt="image">
+
+### 🕵Detecting Visual Bugs Using Terminal
+
+- run command and wait for the results;
+
+```
+npx playwright test
+```
+
+- if there are any problems whith screenshots comparison, go to `src/__tests__/visualTesting/results` and watch the results.
+
+### 🔁Update Screenshots
+
+When intentional design changes cause test failures, to create new references use one of the following ways:
+
+- update all the screenshots across the project using
+
+```
+npx playwright test --update-snapshots
+```
+
+- to update one or several screenshots only, please delete outdated reference images from the according folder (`src/__tests__/visualTesting/[NAME_OF_THE_TEST].spec.ts-snapshots`) and rerun tests again. On testing without samples, Playwright firstly will create them from your code.
+
+### 👀 Real-time Browsing via Specified Engine
+
+You can even browse the websites using current engine. To do that, please use command:
+
+```
+npx playwright open -b webkit rs.school
```

Comment:
Wouldn't it be better to specify a local address here?

Reply:
It is just for example. I hope developers will replace server with one they need.

Reply:
Specifying local server is more confusing. We don't know wether the local server is running or is it using the same port. And then for those, who are "in the tank" we should add the instruction to run local service firstly and make sure than it has a `5173` port and so on. I think this command example is the best instruction.
But you can order me to do so as a code owner)